lib/_http_agent.js

```diff
@@ -115,6 +115,9 @@ Agent.prototype.addRequest = function(req, options) {
     };
   }
 
+  options = util._extend({}, options);
```

Comment:
So can we keep the same syntax throughout the file (_extends) and refactor it to a `Object.assign` at a future PR?